### ISSUE TITLE
0.0.17+ l2ethers deploy behaviour is different to 0.0.16

### ISSUE THREAD

**Issue by azf20** [ethereum-optimism/plugins#23]
To reproduce:

Download & install node_modules, generate deployer account
```
git clone -b local-optimism https://github.com/austintgriffith/scaffold-eth.git local-optimism
cd local-optimism
yarn install
yarn generate
```

In another terminal spin up local Optimism:
```
cd docker/optimism-integration
git submodule init
git submodule update
make up
```

Run test deploy script:
```
yarn workspace @scaffold-eth/hardhat hardhat run scripts/simple-oe-deploy.js
```
**Should deploy successfully, and print code**

- Upgrade `"@eth-optimism/plugins"` in  `packages/hardhat/package.json` to 0.0.19
- Edit YourContract.sol in `packages/hardhat/contracts` so that it is recompiled

Re-run test deploy script:
```
yarn workspace @scaffold-eth/hardhat hardhat run scripts/simple-oe-deploy.js
```
Deploy appears to succeed, but there is no code at the deployed address:
 📄 YourContract deployed to: 0x5FbDB2315678afecb367f032d93F642f64180aa3 (0.076 ETH)
0x

**Comment by smartcontracts** [comment]
Looking into this now. Thanks for the report!

**Comment by azf20** [comment]
ok nice this deployment issue is fixed in 0.0.20 ❤️ 

However I now see a strange issue in my E2E hardhat demo (full instructions in the repo README, it requires funding of the deployer account on L1..):
`yarn deploy-oe`
The contracts all deploy fine, as does the deposit from L1->L2, but then functions involving a transfer on `OVM_L2DepositedERC20` seem to fail in 0.0.20 (which go through on 0.0.16 - `withdraw`, `transfer`). Other functions on the same contract work as expected (e.g. `approve`). I don't think it's a contract issue, as they work end-to-end as expected in the frontend (using regular `ethers`).
This is probably separate to the original issue above, but wondered if there were any obvious connected changes?

**Comment by smartcontracts** [comment]
Sorry for the lack of a response here. I totally missed your comment. Going to close this issue because the original parent problem should be obsolete now (`l2ethers` is no longer exposed). If you're still having any other issues with the plugins plz let me know and I'll try to get them fixed up!!
